pkg/cmd/roachtest/test_runner.go

```diff
@@ -605,17 +605,24 @@ func (r *testRunner) runWorker(
 			qp.Release(alloc)
 		}
 	}()
+	// cluster destroy can be done concurrently. We just need to make sure that we exit the roachtest once all
+	// the cluster destroy operations are complete!
+	clusterDestroyWg := &sync.WaitGroup{}
 
 	// Loop until there's no more work in the pool, we get interrupted, or an
 	// error occurs.
 	for {
 		select {
 		case <-interrupt:
 			l.ErrorfCtx(ctx, "worker detected interruption")
+			r.destroyCluster(ctx, clusterDestroyWg, c, l, true)
+			clusterDestroyWg.Wait() // wait for the clusters to be destroyed
 			return errors.Errorf("interrupted")
 		default:
 			if ctx.Err() != nil {
 				// The context has been canceled. No need to continue.
+				r.destroyCluster(ctx, clusterDestroyWg, c, l, true)
+				clusterDestroyWg.Wait() // wait for the clusters to be destroyed
```

Comment:
It seems like the `waitgroup wait` is added before all possible return points. Is there any reason for not moving it a `defer` above, and creating the waitgroup a little earlier so it can be part of that `defer`, so if someone adds another return point they wouldn't have to keep this in mind? Similar to how the `quota` is release.